qemu/tests/virtio_serial_hotplug_port_pci.py

```diff
@@ -34,6 +34,39 @@ def get_buses_and_serial_devices(vm, params, char_devices, serials):
     return buses, serial_devices
 
 
+@error_context.context_aware
+def reboot_guest(test, params, vm, session):
+    """
+    Reboot guest from system_reset or shell.
+    """
+
+    vm.reboot(session, method=params["reboot_method"])
+
+
+@error_context.context_aware
+def shutdown_guest(test, params, vm, session):
+    """
+    Shutdown guest via system_powerdown or shell.
+    """
+
+    if params.get("shutdown_method") == "shell":
+        session.sendline(params["shutdown_command"])
+    elif params.get("shutdown_method") == "system_powerdown":
+        vm.monitor.system_powerdown()
+    if not vm.wait_for_shutdown(int(params.get("shutdown_timeout", 360))):
+        test.fail("guest refuses to go down")
+
+
+@error_context.context_aware
+def live_migration_guest(test, params, vm, session):
+    """
+    Run migrate_set_speed, then migrate guest.
+    """
+
+    qemu_migration.set_speed(vm, params.get("mig_speed", "1G"))
+    vm.migrate()
+
+
```

Comment:
Sorry, I don't see the point why we need to implement them again here. Why not `from vioser_in_use import live_migration_guest`?

Reply:
what message I got from the maintainer is better to avoid calling case by case.

Reply:
@6-dehan This is not calling case, just importing independent functions from other module, calling case means you call the `run()` function in another module. Please help at the maintainer here, we could discuss on it.

Reply:
`globals().get(interrupt_test)(test, params, vm, session)` Here is the "calling case by case", if you want to avoid it, this is what you should re-work on. Currently modification of moving those independent functions does not resolve anything.

Reply:
Though, all the functions are steps, e.g. vm.migrate(), vm.reboot(), those are steps instead of a 'case', the test scenario requires this combined steps, I don't see how we can implement it another way.